pandas/tseries/offsets.py

```diff
@@ -721,43 +778,52 @@ def apply(self, other):
                 skip_bd = BusinessDay(n=bd)
                 # midnight business hour may not on BusinessDay
                 if not self.next_bday.onOffset(other):
-                    remain = other - self._prev_opening_time(other)
-                    other = self._next_opening_time(other + skip_bd) + remain
+                    prev_open = self._prev_opening_time(other)
+                    remain = other - prev_open
+                    other = prev_open + skip_bd + remain
                 else:
                     other = other + skip_bd
 
             hours, minutes = divmod(r, 60)
-            result = other + timedelta(hours=hours, minutes=minutes)
+            rem = timedelta(hours=hours, minutes=minutes)
 
             # because of previous adjustment, time will be larger than start
             if n >= 0:
-                bday_edge = self._prev_opening_time(other) + bhdelta
-                if bday_edge < result:
-                    bday_remain = result - bday_edge
-                    result = self._next_opening_time(other)
-                    result += bday_remain
+                while rem != timedelta(0):
```

Comment:
can you add some comments on what you are doing.

can you use more meaningful variable names

Reply:
Updated